server/src/main/java/org/folio/metastorage/server/OaiPmhClientService.java

```diff
@@ -0,0 +1,538 @@
+package org.folio.metastorage.server;
+
+import io.netty.handler.codec.http.QueryStringEncoder;
+import io.vertx.core.Future;
+import io.vertx.core.MultiMap;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.RequestOptions;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.validation.RequestParameters;
+import io.vertx.ext.web.validation.ValidationHandler;
+import io.vertx.sqlclient.Row;
+import io.vertx.sqlclient.RowIterator;
+import io.vertx.sqlclient.SqlConnection;
+import io.vertx.sqlclient.Tuple;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.metastorage.oai.OaiParserStream;
+import org.folio.metastorage.oai.OaiRecord;
+import org.folio.metastorage.util.SourceId;
+import org.folio.metastorage.util.XmlMetadataParserMarcInJson;
+import org.folio.metastorage.util.XmlMetadataStreamParser;
+import org.folio.metastorage.util.XmlParser;
+import org.folio.okapi.common.HttpResponse;
+
+public class OaiPmhClientService {
+
+  Vertx vertx;
+
+  HttpClient httpClient;
+
+  private static final String STATUS_LITERAL = "status";
+
+  private static final String RESUMPTION_TOKEN_LITERAL = "resumptionToken";
+
+  private static final String IDLE_LITERAL = "idle";
+
+  private static final String RUNNING_LITERAL = "running";
+
+  private static final String CONFIG_LITERAL = "config";
+
+  private static final String TOTAL_RECORDS_LITERAL = "totalRecords";
+
+  private static final String TOTAL_REQUESTS_LITERAL = "totalRequests";
+
+  private static final String B_WHERE_ID1_LITERAL = " WHERE ID = $1";
```

Comment:
Do we need to use those literal? It makes the code harder to read.

Reply:
Thx to sq. I can try to make them constants again